### DISCUSSION TITLE
Admin Component library

### DISCUSSION THREAD

**Discussion by olivermrbl** [medusajs/medusa#2855] [Feature Requests]
### What
Introduce a Medusa Admin component library, that will allow plugin-builders to create components at ease.

**Comment by abusarah-tech** [comment]
Is this going to be using Storybook for component visualisation? I would appreciate to contribute once its ready for the community.

**Reply by olivermrbl** [reply]
Yes, we've set up Storybook: https://storybook.medusajs.com/?path=/story/components-alertdialog--default

The repository is currently private, but we expect to open-source it very, very soon. We just need to do some house-keeping :)

**Reply by abusarah-tech** [reply]
Great work! That's looking pretty good.

**Comment by pevey** [comment]
My hope would be that it continues to use radix at it's core.

**Comment by pevey** [comment]
I'd like to request a Table component.  @kasperkristensen

**Reply by olivermrbl** [reply]
The Table component is in the making. You can expect it in a couple of months (due to holiday season) :)

**Comment by olivermrbl** [comment]
### Update: Alpha version has been released!

We are very excited to share the first alpha version of our new admin component library. Implementation led by @kasperkristensen and @StephixOne. 

The component library is yet to be documented, so you'll have to explore it yourself for now. Though, we've deployed a Storybook preview allowing you to play around with the components and their controls (most of them): https://storybook.medusajs.com/

Also, all complex components are built on top of Radix UI, so if you are familiar with their primitives, it should be fairly easy to pick up on.

To get started, you will need three packages:
```
@medusajs/admin@alpha
@medusajs/ui@alpha
@medusajs/icons@alpha // this one is optional - only needed if you want to use icons
```

If you are working on a project, that has used @medusajs/admin previously, you might need to clear `node_modules`. 

Please note, this is an alpha version, so the component API and props are likely to change in the near future.

Aside from the components included in this first iteration, we are working on a Table and Select.
